inst/tests/tests.Rraw

```diff
@@ -18315,3 +18315,45 @@ if (test_bit64) {
   apple = data.table(id = c("a", "b", "b"), time = c(1L, 1L, 2L), y = i64v[1:3])
   test(2248, dcast(apple, id ~ time, value.var = "y"), data.table(id = c('a', 'b'), `1` = i64v[1:2], `2` = i64v[4:3], key='id'))
 }
+
+# trunc.char misbehaves for multibyte characters, #5096
+# Tests for combining characters
+options(datatable.prettyprint.char=4L)
+accented_a = "\u0061\u0301"
+ja_ichi = "\u4E00"
+ja_ni = "\u4E8C"
+ja_ko = "\u3053"
+ja_n = "\u3093"
+halfwidth_bracket = "\uFF63"
+dots_regex = "[.][.][.]"
+DT = data.table(strrep(accented_a, 4L))
+test(2248.01, DT, output=paste("1:", strrep(accented_a, 4L)))
+options(datatable.prettyprint.char=3L)
+test(2248.02, DT, output=paste0("1: ", accented_a, accented_a, accented_a, dots_regex))
+options(datatable.prettyprint.char=1L)
+test(2248.03, DT, output=paste0("1: ", accented_a, dots_regex))
+# Tests for full-width multibyte characters
+options(datatable.prettyprint.char=4L)
+# four Japanese characters: 'one' 'two' 'ko' 'n'
+DT = data.table(paste0(ja_ichi, ja_ni, ja_ko, ja_n))
+test(2248.04, DT, output=paste0("1: ", ja_ichi, ja_ni, ja_ko, ja_n))
+options(datatable.prettyprint.char=3L)
+test(2248.05, DT, output=paste0("1: ", ja_ichi, ja_ni, ja_ko, dots_regex))
+options(datatable.prettyprint.char=1L)
+test(2248.06, DT, output=paste0("1: ", ja_ichi, dots_regex))
+# Multiple tests, all truncated
+DT = data.table(strrep(accented_a, 4L), strrep(ja_ichi, 4L), strrep("a", 4L), strrep(halfwidth_bracket, 4L))
+options(datatable.prettyprint.char=2L)
+test(2248.07, DT, output=paste0("1: ", accented_a, accented_a, dots_regex, " ", ja_ichi, ja_ichi, dots_regex, " aa", dots_regex, " ", halfwidth_bracket, halfwidth_bracket, dots_regex))
+options(datatable.prettyprint.char=3L)
+test(2248.08, DT, output=paste0("1: ", accented_a, accented_a, accented_a, dots_regex, " ", ja_ichi, ja_ichi, ja_ichi, dots_regex, " aaa", dots_regex, " ", halfwidth_bracket, halfwidth_bracket, dots_regex))
```

Comment:
@joshhwuu this test is failing, do we need another `halfwidth_bracket`?

```
  Expected: <<1: ááá[.][.][.] 一一一[.][.][.] aaa[.][.][.] ｣｣[.][.][.]>>
  Observed: <<       V1        V2     V3     V4\n1: ááá... 一一一... aaa... ｣｣｣...>>
```

Reply:
Yes, sorry there should be three `halfwidth_brackets`.

Reply:
OK, ready to merge after that